server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchFieldsPhase.java

```diff
@@ -81,9 +99,12 @@ public FetchSubPhaseProcessor getProcessor(FetchContext fetchContext) {
                     }
                 }
             }
+            metadataFields.addAll(fetchContextMetadataFields);
             metadataFieldFetcher = FieldFetcher.create(searchExecutionContext, metadataFields);
         } else {
-            metadataFieldFetcher = FieldFetcher.create(searchExecutionContext, DEFAULT_METADATA_FIELDS);
+            final Set<FieldAndFormat> allMetadataFields = new HashSet<>(DEFAULT_METADATA_FIELDS);
```

Comment:
What is the reason behind this change?

Reply:
This is required to be able to fetch some stored metadata fields, like `ignored_source` when tehy are requested via `fields`. Normally such fields would be requested via `stored_fields`. After the latest changes (done like 6 months ago) the decision was to fetch fields using the `ValueFetcher` interface which means we do not distinguish anymore stored versus non-stored. Anyway the idea, as you can see from this code is to use the thise part of the logic to fetch metadata fields. In this case we need to do this to fetch metadata fields requested via `fields` (and not via `stored_fields`).